File: backend/utils/seed.py
from datetime import datetime, timedelta
import bcrypt

async def seed_database(db):
    """Seed database with initial data"""
    
    # Seed admin user
    admin_exists = await db.admin_users.find_one({"email": "admin@havosec.com"})
    if not admin_exists:
        hashed_password = bcrypt.hashpw("admin123".encode(), bcrypt.gensalt()).decode()
        await db.admin_users.insert_one({
            "email": "admin@havosec.com",
            "password": hashed_password,
            "firstName": "Admin",
            "lastName": "User",
            "role": "admin",
            "permissions": ["all"],
            "createdAt": datetime.utcnow()
        })
        print("[OK] Admin user seeded")
    
    # Seed client user
    client_exists = await db.users.find_one({"email": "client@havosec.com"})
    if not client_exists:
        hashed_password = bcrypt.hashpw("client123".encode(), bcrypt.gensalt()).decode()
        await db.users.insert_one({
            "email": "client@havosec.com",
            "password": hashed_password,
            "firstName": "Client",
            "lastName": "User",
            "company": "HavoSec",
            "role": "viewer",
            "isActive": True,
            "emailVerified": True,
            "createdAt": datetime.utcnow()
        })
        print("[OK] Client user seeded")
    
    # Seed header content
    header_exists = await db.website_content.find_one({"section": "header"})
    if not header_exists:
        await db.website_content.insert_one({
            "section": "header",
            "content": {
                "navLinks": [
                    {"name": "Home", "path": "/"},
                    {"name": "About", "path": "/about"},
                    {"name": "Blog", "path": "/blog"}
                ]
            },
            "updatedAt": datetime.utcnow()
        })
        print("[OK] Header content seeded")
    
    # Seed footer content
    footer_exists = await db.website_content.find_one({"section": "footer"})
    if not footer_exists:
        await db.website_content.insert_one({
            "section": "footer",
            "content": {
                "sections": [
                    {
                        "title": "Product",
                        "links": [
                            {"name": "Features", "path": "/"},
                            {"name": "Security", "path": "/about"},
                            {"name": "Integrations", "path": "#"},
                            {"name": "Pricing", "path": "#"},
                            {"name": "API", "path": "#"}
                        ]
                    },
                    {
                        "title": "Solutions",
                        "links": [
                            {"name": "Enterprise", "path": "#"},
                            {"name": "Small Business", "path": "#"},
                            {"name": "Government", "path": "#"},
                            {"name": "Healthcare", "path": "#"},
                            {"name": "Finance", "path": "#"}
                        ]
                    },
                    {
                        "title": "Resources",
                        "links": [
                            {"name": "Blog", "path": "/blog"},
                            {"name": "Documentation", "path": "#"},
                            {"name": "Help Center", "path": "#"},
                            {"name": "Case Studies", "path": "#"},
                            {"name": "Webinars", "path": "#"}
                        ]
                    },
                    {
                        "title": "Company",
                        "links": [
                            {"name": "About Us", "path": "/about"},
                            {"name": "Careers", "path": "#"},
                            {"name": "News", "path": "#"},
                            {"name": "Contact", "path": "#"},
                            {"name": "Partners", "path": "#"}
                        ]
                    }
                ],
                "socialLinks": [
                    {"name": "LinkedIn", "url": "#", "icon": "linkedin"},
                    {"name": "Twitter", "url": "#", "icon": "twitter"},
                    {"name": "GitHub", "url": "#", "icon": "github"}
                ]
            },
            "updatedAt": datetime.utcnow()
        })
        print("[OK] Footer content seeded")
    
#     # Seed website content
#     sections = ["hero", "features", "about", "services", "testimonials"]
#     for section in sections:
#         exists = await db.website_content.find_one({"section": section})
#         if not exists:
#             content = get_default_content(section)
#             await db.website_content.insert_one({
#                 "section": section,
#                 "content": content,
#                 "updatedAt": datetime.utcnow()
#             })
#     print("[OK] Website content seeded")
    
#     # Seed sample blog posts
#     blog_count = await db.blog_posts.count_documents({})
#     if blog_count == 0:
#         posts = [
#             {
#                 "title": "Advanced Threat Detection with AI",
#                 "slug": "advanced-threat-detection-ai",
#                 "content": "<p>Artificial Intelligence is revolutionizing how we detect and respond to cyber threats...</p>",
#                 "excerpt": "Learn how AI is transforming cybersecurity threat detection.",
#                 "author": "HavoSec Team",
#                 "category": "Technology",
#                 "tags": ["AI", "Threat Detection", "Security"],
#                 "status": "published",
#                 "createdAt": datetime.utcnow(),
#                 "updatedAt": datetime.utcnow()
#             },
#             {
#                 "title": "Zero Trust Architecture Guide",
#                 "slug": "zero-trust-architecture-guide",
#                 "content": "<p>Implementing zero trust security requires a fundamental shift in thinking...</p>",
#                 "excerpt": "A comprehensive guide to implementing zero trust security.",
#                 "author": "HavoSec Team",
#                 "category": "Best Practices",
#                 "tags": ["Zero Trust", "Architecture", "Security"],
#                 "status": "published",
#                 "createdAt": datetime.utcnow(),
#                 "updatedAt": datetime.utcnow()
#             },
#             {
#                 "title": "Ransomware Prevention Strategies",
#                 "slug": "ransomware-prevention-strategies",
#                 "content": "<p>Protecting your organization from ransomware requires multiple layers of defense...</p>",
#                 "excerpt": "Essential strategies to protect against ransomware attacks.",
#                 "author": "HavoSec Team",
#                 "category": "Prevention",
#                 "tags": ["Ransomware", "Prevention", "Security"],
#                 "status": "published",
#                 "createdAt": datetime.utcnow(),
#                 "updatedAt": datetime.utcnow()
#             }
#         ]
#         await db.blog_posts.insert_many(posts)
#         print("[OK] Blog posts seeded")
    
#     # Seed security events for client dashboard
#     events_count = await db.security_events.count_documents({})
#     if events_count == 0:
#         events = generate_security_events()
#         await db.security_events.insert_many(events)
#         print("[OK] Security events seeded")

# def get_default_content(section):
#     defaults = {
#         "hero": {
#             "title": "Secure Your Digital Assets with HavoSec",
#             "subtitle": "Advanced cybersecurity analytics and threat detection platform designed for modern organizations.",
#             "ctaText": "Get Started",
#             "ctaLink": "/book-demo"
#         },
#         "features": {
#             "title": "Advanced Security Features",
#             "items": [
#                 {"title": "Real-Time Threat Detection", "description": "AI-powered threat detection that identifies attacks in milliseconds."},
#                 {"title": "Advanced Monitoring", "description": "Comprehensive visibility across your entire digital infrastructure."},
#                 {"title": "Automated Response", "description": "Instant automated responses to security threats."},
#                 {"title": "Analytics Dashboard", "description": "Rich analytics and reporting for security insights."}
#             ]
#         },
#         "about": {
#             "title": "About HavoSec",
#             "description": "HavoSec is a leading cybersecurity company dedicated to protecting organizations from evolving threats.",
#             "mission": "Our mission is to provide world-class security solutions that enable businesses to operate safely in the digital age."
#         },
#         "services": {
#             "title": "Our Services",
#             "items": [
#                 {"title": "Threat Detection", "description": "24/7 monitoring and threat detection services."},
#                 {"title": "Incident Response", "description": "Rapid response to security incidents."},
#                 {"title": "Security Consulting", "description": "Expert security consulting and advisory."}
#             ]
#         },
#         "testimonials": {
#             "title": "What Our Clients Say",
#             "items": []
#         }
#     }
#     return defaults.get(section, {})

# def generate_security_events():
#     import random
#     events = []
#     event_types = ['attack_blocked', 'intrusion_attempt', 'malware_detected', 'phishing_blocked', 'ddos_mitigated', 'vulnerability_scan']
#     severities = ['low', 'medium', 'high', 'critical']
#     statuses = ['detected', 'blocked', 'investigating', 'resolved']
#     countries = ['US', 'CN', 'RU', 'KP', 'IR', 'TR', 'BR', 'IN']
#     descriptions = [
#         'Suspicious login attempt detected',
#         'SQL injection attack blocked',
#         'Malware signature detected in upload',
#         'DDoS attack mitigated',
#         'Brute force attack on admin panel',
#         'Phishing attempt blocked',
#         'Vulnerability scan detected'
#     ]
    
#     for i in range(100):
#         random_days = random.randint(0, 30)
#         random_hours = random.randint(0, 23)
#         event_time = datetime.utcnow() - timedelta(days=random_days, hours=random_hours)
        
#         events.append({
#             "eventType": random.choice(event_types),
#             "severity": random.choice(severities),
#             "status": random.choice(statuses),
#             "description": random.choice(descriptions),
#             "source": {
#                 "ip": f"{random.randint(1,255)}.{random.randint(1,255)}.{random.randint(1,255)}.{random.randint(1,255)}",
#                 "country": random.choice(countries)
#             },
#             "target": {
#                 "endpoint": f"/api/{random.choice(['login', 'dashboard', 'admin', 'upload', 'search'])}",
#                 "service": random.choice(['web', 'api', 'database', 'auth']),
#                 "port": random.choice([80, 443, 22, 3306])
#             },
#             "tags": ['automated'],
#             "createdAt": event_time,
#             "updatedAt": event_time
#         })
    
#     return events

File: backend/seed.js
const AdminUser = require('./models/AdminUser');
const WebsiteContent = require('./models/WebsiteContent');
const BlogPost = require('./models/BlogPost');

const defaultContent = {
  hero: {
    title: 'Secure Your Digital Assets with',
    subtitle: 'HavoSec',
    description: 'AI-powered cybersecurity platform with Offensive and Defensive models. Real-time threat detection, automated response, and comprehensive security analytics for modern enterprises.',
    primaryButtonText: 'Book a Demo',
    primaryButtonLink: '/book-demo',
    secondaryButtonText: 'Learn More',
    secondaryButtonLink: '/about'
  },
  achievements: {
    items: [
      { id: 1, number: '50M+', label: 'Threats Blocked Daily' },
      { id: 2, number: '99.9%', label: 'Uptime Guarantee' },
      { id: 3, number: '500+', label: 'Enterprise Clients' },
      { id: 4, number: '24/7', label: 'Security Monitoring' }
    ]
  },
  features: {
    title: 'Advanced Security Features',
    features: [
      { id: 1, icon: 'ShieldCheckIcon', title: 'Real-Time Threat Detection', description: 'AI-powered detection identifies and blocks attacks instantly.', benefit: 'Reduce incidents by 95%' },
      { id: 2, icon: 'EyeIcon', title: 'Advanced Monitoring', description: 'Complete visibility across your digital infrastructure.', benefit: 'Zero blind spots' },
      { id: 3, icon: 'BoltIcon', title: 'Automated Response', description: 'Instant automated responses to security threats.', benefit: 'Response under 10 seconds' },
      { id: 4, icon: 'ChartBarIcon', title: 'Analytics Dashboard', description: 'Rich analytics for security insights and compliance.', benefit: 'Data-driven decisions' },
      { id: 5, icon: 'ExclamationTriangleIcon', title: 'Incident Management', description: 'Streamlined incident response workflows.', benefit: '80% faster resolution' },
      { id: 6, icon: 'ClockIcon', title: '24/7 Monitoring', description: 'Round-the-clock security monitoring and threat hunting.', benefit: 'Never miss a threat' }
    ]
  },
  offensive_model: {
    title: 'Offensive Security Model',
    description: 'Our AI-powered offensive model proactively attacks your systems to find vulnerabilities before hackers do.',
    features: [
      'Automated Penetration Testing',
      'Vulnerability Assessment',
      'Red Team Simulations',
      'Attack Surface Mapping',
      'Exploit Detection'
    ]
  },
  defensive_model: {
    title: 'Defensive Security Model',
    description: 'Real-time protection that defends your assets against all types of cyber threats.',
    features: [
      'Intrusion Detection & Prevention',
      'Threat Intelligence',
      'Incident Response',
      'Security Monitoring',
      'Compliance Management'
    ]
  },
  services: {
    title: 'Security Services',
    services: [
      { id: 1, name: 'Web Security', description: 'Comprehensive web application security testing and protection.', icon: 'GlobeAltIcon' },
      { id: 2, name: 'App Security', description: 'Mobile and desktop application security assessment.', icon: 'DevicePhoneMobileIcon' },
      { id: 3, name: 'Network Security', description: 'Network infrastructure security and monitoring.', icon: 'ServerIcon' },
      { id: 4, name: 'Cloud Security', description: 'Cloud infrastructure security and compliance.', icon: 'CloudIcon' },
      { id: 5, name: 'API Security', description: 'API security testing and protection.', icon: 'CodeBracketIcon' },
      { id: 6, name: 'IoT Security', description: 'Internet of Things device security.', icon: 'CpuChipIcon' }
    ]
  },
  about_hero: {
    title: 'About HavoSec',
    description: 'Protecting organizations with AI-powered cybersecurity since 2020'
  },
  about_mission: {
    title: 'Our Mission',
    text: 'To empower organizations with advanced cybersecurity that transforms how they detect, respond to, and prevent cyber threats.'
  },
  about_vision: {
    title: 'Our Vision',
    text: 'To be the global leader in AI-powered cybersecurity, creating a safer digital world.'
  },
  about_company_info: {
    founded: '2020',
    teamSize: '50-100 employees',
    locations: ['San Francisco', 'New York', 'London']
  },
  about_values: {
    values: [
      { id: 1, icon: 'ShieldCheckIcon', title: 'Security First', description: 'Security is our foundation.' },
      { id: 2, icon: 'LightBulbIcon', title: 'Innovation', description: 'Pushing cybersecurity boundaries.' },
      { id: 3, icon: 'EyeIcon', title: 'Transparency', description: 'Clear communication builds trust.' },
      { id: 4, icon: 'HeartIcon', title: 'Customer Success', description: 'Your security is our success.' }
    ]
  },
  demo_cta: {
    title: 'Ready to Secure Your Organization?',
    description: 'Get a personalized demo of HavoSec and see our Offensive and Defensive models in action.',
    buttonText: 'Schedule Your Demo',
    buttonLink: '/book-demo'
  },
  footer: {
    companyName: 'HavoSec',
    tagline: 'AI-Powered Cybersecurity',
    copyrightText: '© 2025 HavoSec. All rights reserved.'
  }
};

const defaultBlogPosts = [
  {
    title: 'Understanding Offensive vs Defensive Security Models',
    slug: 'offensive-vs-defensive-security',
    excerpt: 'Learn how combining offensive and defensive security approaches creates comprehensive protection.',
    content: '<p>Modern cybersecurity requires both offensive and defensive strategies...</p>',
    featuredImage: 'https://images.unsplash.com/photo-1555949963-aa79dcee981c?w=800',
    author: { name: 'HavoSec Team' },
    category: 'cybersecurity',
    tags: ['Offensive Security', 'Defensive Security', 'Best Practices'],
    status: 'published',
    publishedAt: new Date()
  },
  {
    title: 'AI in Threat Detection: The Future of Cybersecurity',
    slug: 'ai-threat-detection-future',
    excerpt: 'How artificial intelligence is revolutionizing threat detection and response.',
    content: '<p>AI-powered threat detection represents the future of cybersecurity...</p>',
    featuredImage: 'https://images.unsplash.com/photo-1563986768609-322da13575f3?w=800',
    author: { name: 'HavoSec Research' },
    category: 'ai',
    tags: ['AI', 'Machine Learning', 'Threat Detection'],
    status: 'published',
    publishedAt: new Date()
  }
];

const seedDatabase = async () => {
  try {
    // Create super admin
    const existingAdmin = await AdminUser.findOne({ email: 'admin@havosec.com' });
    if (!existingAdmin) {
      const admin = new AdminUser({
        email: 'admin@havosec.com',
        password: 'HavoSec@2025',
        firstName: 'Super',
        lastName: 'Admin',
        role: 'super_admin',
        permissions: ['content', 'blog', 'users', 'clients', 'demo_requests', 'settings', 'analytics']
      });
      await admin.save();
      console.log('\x1b[33m%s\x1b[0m', '✓ Default admin created: admin@havosec.com / HavoSec@2025');
    }

    // Seed website content
    const existingContent = await WebsiteContent.countDocuments();
    if (existingContent === 0) {
      for (const [section, content] of Object.entries(defaultContent)) {
        await WebsiteContent.create({ section, content, isActive: true });
      }
      console.log('\x1b[33m%s\x1b[0m', '✓ Website content seeded');
    }

    // Seed blog posts
    const existingPosts = await BlogPost.countDocuments();
    if (existingPosts === 0) {
      for (const post of defaultBlogPosts) {
        await BlogPost.create(post);
      }
      console.log('\x1b[33m%s\x1b[0m', '✓ Blog posts seeded');
    }
  } catch (error) {
    console.error('Seed error:', error);
  }
};

module.exports = { seedDatabase };
